Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/SqlDialect.cs

```diff
@@ -50,5 +53,65 @@ public virtual string DelimitIdentifier (string identifier)
       return "[" + identifier + "]";
     }
 
+    public IDbDataParameter CreateDataParameter (IDbCommand command, IStorageTypeInformation storageTypeInformation, string parameterName, object? value)
+    {
+      ArgumentUtility.CheckNotNull("command", command);
+      ArgumentUtility.CheckNotNull("storageTypeInformation", storageTypeInformation);
+      ArgumentUtility.CheckNotNullOrEmpty("parameterName", parameterName);
+
+      var convertedValue = storageTypeInformation.ConvertToStorageType(value);
+
+      var parameter = command.CreateParameter();
+      parameter.ParameterName = GetParameterName(parameterName);
+      parameter.Value = convertedValue;
+      parameter.DbType = storageTypeInformation.StorageDbType;
+      if (storageTypeInformation.StorageTypeLength.HasValue)
+      {
+        var parameterSize = storageTypeInformation.StorageTypeLength.Value;
+        if (parameterSize == SqlStorageTypeInformationProvider.StorageTypeLengthRepresentingMax)
+        {
+          var fulltextCompatibleMaxLength = storageTypeInformation.StorageDbType switch
+          {
+              DbType.AnsiString => 8000,
+              DbType.String => 4000,
+              // AnsiStringFixedLength and StringFixedLength represent char(...) and nchar(...), which do not support "max" as size
+              _ => default(int?)
+          };
+
+          if (fulltextCompatibleMaxLength.HasValue)
+          {
+            var hasFullTextCompatibleMaxLength = convertedValue switch
+            {
+                DBNull => true,
+                string stringValue => stringValue.Length <= fulltextCompatibleMaxLength.Value,
+                char[] charsValue => charsValue.Length <= fulltextCompatibleMaxLength.Value,
+                _ => false
```

Comment:
I switched this to true, no test was red

Reply:
Changed to exception since this case is theoretically impossible as long as the StorageTypeInfo is correct